octodns/provider/cloudflare.py

```diff
@@ -66,17 +66,24 @@ class CloudflareProvider(BaseProvider):
     MIN_TTL = 120
     TIMEOUT = 15
 
-    def __init__(self, id, email, token, cdn=False, *args, **kwargs):
+    def __init__(self, id, email=None, token=None, cdn=False, *args, **kwargs):
         self.log = getLogger('CloudflareProvider[{}]'.format(id))
         self.log.debug('__init__: id=%s, email=%s, token=***, cdn=%s', id,
                        email, cdn)
         super(CloudflareProvider, self).__init__(id, *args, **kwargs)
 
         sess = Session()
-        sess.headers.update({
-            'X-Auth-Email': email,
-            'X-Auth-Key': token,
-        })
+        if email and token:
+            sess.headers.update({
+                'X-Auth-Email': email,
+                'X-Auth-Key': token,
+            })
+        else:
+            # https://api.cloudflare.com/#getting-started-requests
+            # https://tools.ietf.org/html/rfc6750#section-2.1
+            sess.headers.update({
+                'Authorization': 'Bearer %s'.format(token),
```

Comment:
I received this error when running with `token: env/CLOUDFLARE_TOKEN` and `CLOUDFLARE_TOKEN` set to `Read Zone.Zone` and `Edit Zone.DNS`.

```text
Traceback (most recent call last):
  File "/usr/local/bin/octodns-sync", line 8, in <module>
    sys.exit(main())
  File "/usr/local/lib/python2.7/site-packages/octodns/cmds/sync.py", line 39, in main
    dry_run=not args.doit, force=args.force)
  File "/usr/local/lib/python2.7/site-packages/octodns/manager.py", line 301, in sync
    plans = [p for f in futures for p in f.result()]
  File "/usr/local/lib/python2.7/site-packages/octodns/manager.py", line 56, in result
    return self.func(*self.args, **self.kwargs)
  File "/usr/local/lib/python2.7/site-packages/octodns/manager.py", line 237, in _populate_and_plan
    plan = target.plan(zone)
  File "/usr/local/lib/python2.7/site-packages/octodns/provider/base.py", line 49, in plan
    exists = self.populate(existing, target=True, lenient=True)
  File "/usr/local/lib/python2.7/site-packages/octodns/provider/cloudflare.py", line 261, in populate
    records = self.zone_records(zone)
  File "/usr/local/lib/python2.7/site-packages/octodns/provider/cloudflare.py", line 214, in zone_records
    zone_id = self.zones.get(zone.name, False)
  File "/usr/local/lib/python2.7/site-packages/octodns/provider/cloudflare.py", line 114, in zones
    resp = self._request('GET', '/zones', params={'page': page})
  File "/usr/local/lib/python2.7/site-packages/octodns/provider/cloudflare.py", line 101, in _request
    raise CloudflareError(resp.json())
octodns.provider.cloudflare.CloudflareError: Invalid request headers
```

Reply:
@mamercad What scopes are necessary for the token?

Reply:
I had to grant zone resources too. See screenshot below
![Screenshot from 2020-02-05 16-31-17](https://user-images.githubusercontent.com/743094/73855878-f7233380-4834-11ea-98a8-a412439432e6.png)

Currently it's not possible to only use a single zone granted in Cloudflare as access to `/zones` in API is restricted.